### PR TITLE
Support workspace versions (defined in workspace Cargo tomls)

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

File: src/project_types/rust.rs [hunk 1]
```diff
@@ -25,32 +25,65 @@ pub fn find_cargo_tomls(repo: &Repository) -> Result<Vec<VersionFile>> {
...
             {
-                debug!("Found Cargo.toml: {:?} (version={})", path, version);
-                returned.push(VersionFile::CargoToml(path.to_owned(), version))
+                debug!("Found Cargo.toml: {path:?} (version={version})");
+                returned.push(VersionFile::CargoToml {
+                    path: path.to_owned(),
+                    version,
+                    is_workspace: false,
+                })
+            } else if let Some(version) = toml
```
Comment (ayalash) [inline src/project_types/rust.rs:46]: What is the difference between this condition and the one on line #33 (the green one)?

File: src/pargit.rs [hunk 2]
```diff
@@ -352,34 +358,27 @@ impl Project {
...
-                        next_version(&existing_version, kind),
-                        None,
-                        Some(prefix),
-                    )
+    fn resolve_release(&self, version_spec: VersionSpec) -> Result<Release> {
+        let version_files = self.get_version_files_to_bump()?;
+        let (new_version, prefix) = match version_spec {
+            VersionSpec::Exact(version) => (version, None),
+            VersionSpec::Bump(bump_kind) => {
+                let (current_version, prefix) = if version_files.is_empty() {
```
Comment (ayalash) [inline src/pargit.rs:366]: I don't understand why we will `bail` if there are version file(s) - non empty

File: src/project_types/rust.rs (hunk 1 above)
Comment (vmalloc) [reply src/project_types/rust.rs:46]: This one checks `workspace.package.version` and the previous one checks `package.version`

File: src/pargit.rs (hunk 2 above)
Comment (ayalash) [reply src/pargit.rs:366]: I read the diff wrong... sorry

**Comment by ayalash** [comment]
LGTM

**Pull request merged**